firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -235,50 +230,62 @@ private void persistFid(String fid) throws FirebaseInstallationsException {
   }
 
   /**
-   * Registers the FID with FIS servers if FID is in UNREGISTERED state.
+   * Registers the FID with FIS servers if FID is in UNREGISTERED state. Also, refreshes auth token
+   * if expired.
    *
    * <p>Updates FID registration status to PENDING to avoid multiple network calls to FIS Servers.
    */
-  private Task<String> registerFidIfNecessary(
-      PersistedFidEntry persistedFidEntry, AwaitListener listener) {
-    String fid = persistedFidEntry.getFirebaseInstallationId();
-
-    // Check if the fid is unregistered
-    if (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.UNREGISTERED) {
-      updatePersistedFidWithPendingStatus(fid);
-      executeFidRegistration(persistedFidEntry, listener);
-    } else {
-      updateAwaitListenerIfRegisteredFid(persistedFidEntry, listener);
-    }
+  private Task<String> registerFidIfNecessary(AwaitListener listener) {
+    synchronized (persistedFid) {
+      PersistedFidEntry persistedFidEntry = persistedFid.readPersistedFidEntryValue();
```

Comment:
can this value be null? you're checking for null from this call in other places, so there prob should be an explanation of why this instance is safe

Reply:
Ideally not ( if empty, in the previous we create and persist fid).  However, to avoid any NPE, added null checks.